src/python/pants/core/subsystems/python_bootstrap.py

```diff
@@ -240,11 +242,16 @@ def get_pyenv_root(env: Environment) -> str | None:
 
 @rule
 async def python_bootstrap(
-    python_bootstrap_subsystem: PythonBootstrapSubsystem, chosen_environment: ChosenLocalEnvironment
+    python_bootstrap_subsystem: PythonBootstrapSubsystem,
+    chosen_environment: ChosenLocalEnvironmentAlias,
```

Comment:
My next PR will instead have this request `ResolvedEnvironmentTarget`, which will be injected for all goal_rules to match the localhost.